python/lsst/analysis/tools/actions/plot/scatterplotWithTwoHists.py

```diff
@@ -732,20 +686,22 @@ def _makeTopHistogram(
         if "galaxies" in self.plotTypes:  # type: ignore
             totalX.append(cast(Vector, data["xGalaxies"]))
         if "unknown" in self.plotTypes:  # type: ignore
-            totalX.append(cast(Vector, data["xUknown"]))
+            totalX.append(cast(Vector, data["xUnknown"]))
         if "any" in self.plotTypes:  # type: ignore
             totalX.append(cast(Vector, data["x"]))
 
+        # This filters out nans, but that probably should have happened earlier
```

Comment:
This should have happened earlier, I don't like taking NaNs out of x and not their corresponding y.

Reply:
It's fine because it's just for histograms. In fact all of the plotting calls will filter out non-finite points, so the `[x for x in totalX if x == x]` lines were unnecessary. The only issues come with trying to set limits without filtering non-finite points (not just nan ones).

I do agree that filtering should happen earlier. Probably most plots should select S/N > 3 or 5, like the scatterplot unit test does.